lib/ansible/parsing/vault/__init__.py

```diff
@@ -656,10 +656,10 @@ def decrypt_and_get_vault_id(self, vaulttext):
         b_vaulttext = to_bytes(vaulttext, nonstring='error')  # enforce vaulttext is str/bytes, keep type check if removing type conversion
 
         if self.secrets is None:
-            raise AnsibleVaultError("A vault password must be specified to decrypt data.", obj=vaulttext)
+            raise AnsibleVaultError("A vault password must be specified to decrypt data.")
 
         if not is_encrypted(b_vaulttext):
-            raise AnsibleVaultError("Input is not vault encrypted data.", obj=vaulttext)
+            raise AnsibleVaultError("Input is not vault encrypted data.")
```

Comment:
Please restore passing the obj parameter in this method. If you have a plaintext file, `ansible-vault edit test.txt` will display the text that couldn't be edited. It also ensures that callers other than ansible-vault will display the file and line number of vaulted variables which can't be decrypted.